File: 3_simulation/background/scripts/addZeroIfSampleNotPresent.rb

samples = []
samples_vcf = []



def first(line)
	return line[0..8].join("\t")
end

File.open(ARGV[0]) do |f|
	samples = f.read.chomp.split("\t")
end

while line = STDIN.gets
	if line.start_with? "##"
		puts line
		next
	end
	line_split = line.chomp.split("\t")
	if line.start_with? "#"
		samples_vcf = line_split[9..line_split.size]
		print first(line_split)+"\t"+samples.join("\t")
		next
	end
	puts
	print first(line_split)
	genotypes = line_split[9..line_split.size]
	i = 0
	while i < samples.size
		if (!samples_vcf.find_index(samples[i]).nil?)
			pos =  samples_vcf.find_index(samples[i])
			print "\t"+genotypes[pos]
		else
			print "\t0"
		end
		i+=1
	end
end

File: 3_simulation/background/scripts/addZeroIfSampleNotPresent.py
import sys

samples = []
samples_vcf = []

def first(line):
	return "\t".join(line[0:9])


f = open(sys.argv[1])
samples = f.read().strip().split("\t")

for line in sys.stdin:

	if line.startswith("##"):
		sys.stdout.write(line)
		continue

	line_split = line.strip().split("\t")

	if line.startswith("#"):
		samples_vcf = line_split[9:len(line_split)]
		sys.stdout.write(first(line_split)+"\t" + "\t".join(samples))
		continue
	
	sys.stdout.write("\n")
	sys.stdout.write(first(line_split))
	genotypes = line_split[9:len(line_split)]
	i = 0

	while i < len(samples):
		if samples[i] in samples_vcf:
			pos =  samples_vcf.index(samples[i])
			sys.stdout.write("\t"+genotypes[pos])
		else:
			sys.stdout.write("\t0")

		i+=1
